File: cpp/PE042.cpp
#include <cstdio>
#include <cmath>
bool isTriangle(int n) {
    if(n == 0)
        return false;
    int t = 2*n;
    int k = (int)sqrt(t);
    if(k*(k+1) == t)
        return true;
    return false;
}
int main() {
    int count = 0, c, sum;
    freopen("PE042.txt", "r", stdin);
    c = getchar();
    while(c != EOF){
        if(c == ',')
             c = getchar();
        sum = 0;
        while((c = getchar()) != '\"'){
            sum += c - 'A' + 1;
        }
        c = getchar();
        if(isTriangle(sum)){
            count ++;
        }
    }
    printf("%d\n", count);
    return 0;
}

File: python/PE042.py
#!/usr/bin/env python
import math
import os

def isTriangle(n):
    t = int((-1+math.sqrt(1+8*n))/2)
    if n == (t*(t+1))/2:
        return True
    return False

def main():
    datafile = os.path.abspath(__file__)[:-2] + 'txt'
    s = open(datafile).readline().split(",")
    count = 0
    for t in s:
        Sum = 0
        for c in t:
            if c >= 'A' and c <= 'Z':
                Sum += ord(c) - ord('A') + 1
        if isTriangle(Sum):
            count += 1
    print count

if __name__ == '__main__':
    main()
